rust/arrow/src/compute/kernels/concat.rs

```diff
@@ -0,0 +1,395 @@
+        DataType::Int16 => concat_raw_primitive::<Int16Type>(array_data_list),
+        DataType::Int32 => concat_raw_primitive::<Int32Type>(array_data_list),
+        DataType::Int64 => concat_raw_primitive::<Int64Type>(array_data_list),
+        DataType::UInt8 => concat_raw_primitive::<UInt8Type>(array_data_list),
+        DataType::UInt16 => concat_raw_primitive::<UInt16Type>(array_data_list),
+        DataType::UInt32 => concat_raw_primitive::<UInt32Type>(array_data_list),
+        DataType::UInt64 => concat_raw_primitive::<UInt64Type>(array_data_list),
+        DataType::Float32 => concat_raw_primitive::<Float32Type>(array_data_list),
+        DataType::Float64 => concat_raw_primitive::<Float64Type>(array_data_list),
+        DataType::Date32(_) => concat_raw_primitive::<Date32Type>(array_data_list),
+        DataType::Date64(_) => concat_raw_primitive::<Date64Type>(array_data_list),
+        DataType::Time32(Second) => {
+            concat_raw_primitive::<Time32SecondType>(array_data_list)
+        }
+        DataType::Time32(Millisecond) => {
+            concat_raw_primitive::<Time32MillisecondType>(array_data_list)
+        }
+        DataType::Time64(Microsecond) => {
+            concat_raw_primitive::<Time64MicrosecondType>(array_data_list)
+        }
+        DataType::Time64(Nanosecond) => {
+            concat_raw_primitive::<Time64NanosecondType>(array_data_list)
+        }
+        DataType::Timestamp(Second, _) => {
+            concat_raw_primitive::<TimestampSecondType>(array_data_list)
+        }
+        DataType::Timestamp(Millisecond, _) => {
+            concat_raw_primitive::<TimestampMillisecondType>(array_data_list)
+        }
+        DataType::Timestamp(Microsecond, _) => {
+            concat_raw_primitive::<TimestampMicrosecondType>(array_data_list)
+        }
+        DataType::Timestamp(Nanosecond, _) => {
+            concat_raw_primitive::<TimestampNanosecondType>(array_data_list)
+        }
+        DataType::Interval(IntervalUnit::YearMonth) => {
+            concat_raw_primitive::<IntervalYearMonthType>(array_data_list)
+        }
+        DataType::Interval(IntervalUnit::DayTime) => {
+            concat_raw_primitive::<IntervalDayTimeType>(array_data_list)
+        }
+        DataType::Duration(TimeUnit::Second) => {
+            concat_raw_primitive::<DurationSecondType>(array_data_list)
+        }
+        DataType::Duration(TimeUnit::Millisecond) => {
+            concat_raw_primitive::<DurationMillisecondType>(array_data_list)
+        }
+        DataType::Duration(TimeUnit::Microsecond) => {
+            concat_raw_primitive::<DurationMicrosecondType>(array_data_list)
+        }
+        DataType::Duration(TimeUnit::Nanosecond) => {
+            concat_raw_primitive::<DurationNanosecondType>(array_data_list)
+        }
+        t => unimplemented!("Concat not supported for data type {:?}", t),
+    }
+}
+
+fn concat_primitive<T>(array_data_list: &Vec<ArrayDataRef>) -> Result<ArrayRef>
```

Comment:
do you mind changing these to `&[ArrayDataRef]`? There was a PR a few weeks ago to change borrowed vecs to slices, I think it's one of the clippy lints

Reply:
good catch. I have updated it.